frontend/src/general/SnackbarMessage.jsx

```diff
@@ -0,0 +1,166 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Snackbar, SnackbarContent, IconButton, Box } from "@material-ui/core";
+import { CheckCircle, Error, Close } from "@material-ui/icons";
+import { withRouter } from "react-router";
+
+let openSnackbarFn;
+
+class SnackbarMessage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: false,
+      isSuccess: false,
+      status: "",
+      message: "",
+      pathname: ""
+    };
+    this.setOpen = this.setOpen.bind(this);
+    this.setIsSuccess = this.setIsSuccess.bind(this);
+    this.setStatus = this.setStatus.bind(this);
+    this.setMessage = this.setMessage.bind(this);
+    this.setPathname = this.setPathname.bind(this);
+    this.addSnackbarProperties = this.addSnackbarProperties.bind(this);
+    this.openSnackbar = this.openSnackbar.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+    this.refresh = this.refresh.bind(this);
+  }
+
+  componentDidMount() {
+    openSnackbarFn = this.openSnackbar;
+  }
+
+  handleClose(event, reason) {
+    if (reason === "clickaway") {
+      return;
+    }
+    this.setOpen(false);
+  }
+
+  setOpen(open) {
+    this.setState({
+      open
+    });
+  }
+
+  setStatus(status) {
+    this.setState({
+      status
+    });
+  }
+
+  setIsSuccess(isSuccess) {
+    this.setState({
+      isSuccess
+    });
+  }
+
+  setMessage(message) {
+    this.setState({
+      message
+    });
+  }
+
+  setPathname(pathname) {
+    this.setState({
+      pathname
+    });
+    return pathname;
+  }
+
+  addSnackbarProperties(stat, msg) {
+    this.setStatus(stat);
+    if (stat !== "error") {
+      this.setIsSuccess(true);
+    } else {
+      this.setIsSuccess(false);
+    }
+    this.setMessage(msg);
+  }
+
+  openSnackbar(stat, msg, path) {
+    const currentPathname = this.setPathname(path);
+    if (currentPathname !== "") {
+      this.refresh();
+      setTimeout(() => {
+        this.addSnackbarProperties(stat, msg);
+        this.setOpen(true);
+      }, 300);
+    } else {
+      this.addSnackbarProperties(stat, msg);
+      this.setOpen(true);
+    }
+  }
+
+  refresh() {
+    const { history, location } = this.props;
+    const { pathname } = this.state;
+    if (location.pathname.endsWith(pathname)) {
+      history.go(0);
+    } else {
+      history.push(`/${pathname}`);
+    }
+  }
```

Comment:
I feel like refreshing and redirecting should not be a responsibility of the snackbar message. The snackbar message should probably only be responsible for displaying the message itself.

With the current design there is a lot of conditional logic here - if pathname is empty do nothing, if it's nonempty but we're on that page then refresh, if it's nonempty and we're not on that page redirect... I feel like it would be simpler if the component that opens the snackbar is the one who has to call `history.push` or `history.go` if they want to.

Reply:
This would lead to a lot of props passing of the useHistory() hook, but it could be possible to wrap all the class components (which is the primary react component type we are using for the project) with the withRouter function from react-router-dom. I wasn't sure if that was better architecturally to do so.

Reply:
Also with the way the snackbar is architectured, I think it would be difficult to have the component handle the refreshing and then calling the snackbar. Originally, before I changed snackbar to appear after refresh, the refresh would be dependent on the snackbar's close button and autoHide. However I'm thinking about if it would be possible to refresh and then know which component displayed the snackbar since the refresh might unmount the component, and with the way snackbar is architectured, it might not even call the snackbar function if the component that got unmounted is no longer existent. But I haven't actually investigated to see if that would happen.

Reply:
After some testing with @felixlapierre we discovered that the refresh can be removed as a responsibility of the snackbar, so the fix that was suggested will be implemented.